src/main/java/com/tagtraum/perf/gcviewer/view/GCViewerGui.java

```diff
@@ -23,12 +25,41 @@ public class GCViewerGui extends JFrame {
     private GCViewerGuiToolBar toolBar;
     private GCPreferences preferences;
 
+    private static boolean isShiftPressed;
+    private static boolean isControlPressed;
+    private static boolean isAltPressed;
+
+    public static boolean isShiftPressed() {
+        return isShiftPressed;
+    }
+
+    public static boolean isControlPressed() {
+        return isControlPressed;
+    }
+
+    public static boolean isAltPressed() {
+        return isAltPressed;
+    }
     private RecentGCResourcesMenu recentResourceNamesMenu;
 
     public GCViewerGui() {
         super("tagtraum industries incorporated - GCViewer");
-
         setDefaultCloseOperation(DO_NOTHING_ON_CLOSE);
+
+        KeyboardFocusManager manager = KeyboardFocusManager.getCurrentKeyboardFocusManager();
+        manager.addKeyEventDispatcher(new KeyEventDispatcher() {
+            @Override
+            public boolean dispatchKeyEvent(KeyEvent e) {
+
+                if (e.getID() == KeyEvent.KEY_PRESSED && e.getKeyCode() == KeyEvent.VK_SHIFT) { isShiftPressed = true; }
+                else if (e.getID() == KeyEvent.KEY_RELEASED && e.getKeyCode() == KeyEvent.VK_SHIFT) { isShiftPressed = false; }
+                else if (e.getID() == KeyEvent.KEY_PRESSED && e.getKeyCode() == KeyEvent.VK_CONTROL) { isControlPressed = true; }
+                else if (e.getID() == KeyEvent.KEY_RELEASED && e.getKeyCode() == KeyEvent.VK_CONTROL) { isControlPressed = false; }
+                else if (e.getID() == KeyEvent.KEY_PRESSED && e.getKeyCode() == KeyEvent.VK_ALT) { isAltPressed = true; }
+                else if (e.getID() == KeyEvent.KEY_RELEASED && e.getKeyCode() == KeyEvent.VK_ALT) { isAltPressed = false; }
```

Comment:
Is there a reason you track button states in the static final fields rather than use `e.getModifiersEx()`?

Reply:
See https://github.com/apache/jmeter/commit/38eb2be8fb147461f47709f2c85e957518099edb#diff-c29d87213ad842384446cdab147eb08eec0e24362862aec545e541a957f4e2d7R247

Reply:
It totally slipped my mind that you could do that from a mousewheel event. I've removed this code and rewritten the ModelChartImpl to use this method instead.